pkg/kubectl/cmd/util/factory.go

```diff
@@ -93,7 +93,7 @@ type ClientAccessFactory interface {
 	ClientSet() (internalclientset.Interface, error)
 
 	// KubernetesClientSet gives you back an external clientset
-	KubernetesClientSet() (*kubernetes.Clientset, error)
+	KubernetesClientSet(requiredVersion *schema.GroupVersion) (*kubernetes.Clientset, error)
```

Comment:
Setting a GroupVersion for a clientset looks more than odd, and reading through the code path behind this confirms me that this dates back in ancient history https://github.com/sttts/kubernetes/blob/7773a30f67cc00719a88289170ba21555f8fc0de/pkg/client/unversioned/helper.go#L34. 

What is the goal here? Can we find a real solution for "give me versioned client by GroupVersion"?

Reply:
With the other changes below, I think we don't make use of this "give me versioned client by GroupVersion". The new code hard-codes the GroupVersion from the clientset instead of using the GroupVersion that comes from the RESTMapper. I wonder whether the discovery based RESTMapper (comapre https://github.com/kubernetes/kubernetes/pull/54493/files#diff-382db7246643fb238d071e21dfedf623L347) makes sense at all after going this route for all commands.

@deads2k ^^

Reply:
Once you switch to external, you don't have to pass a groupversion.

Reply:
@sttts , @deads2k Thanks for the review. I have tried to address your concerns. This is my first time writing code for Kubernetes, so I am not entirely sure if I have addressed the issue properly.